### ISSUE TITLE
Problem with generated pkcs#12

### ISSUE THREAD

**Issue by pedrohls94** [digitalbazaar/forge#425]
Hey!

I am trying to generate a pkcs#12 to be imported in a browser, but there's something wrong with the file I'm creating. Firefox/Chrome say it couldn't be imported bc of an unknown error, and opening it with pk12util tool gives the following error: 

> function failed: SEC_ERROR_LEGACY_DATABASE: The certificate/key database is in an old, unsupported format.

I am using this library to create the keypair, the certificate from a PEM and then the pkcs12.
Here's the code I'm using:

To create the keypair:
`var keys = forge.pki.rsa.generateKeyPair(keySize);`

To get the public key pem and sent it to the server (the certificate is created in an application I'm currently working on):
`var pem = pki.publicKeyToPem(keys.publicKey);`

To store the key and get it on another web page:
`sessionStorage.setItem("letsallprayitworks", JSON.stringify(keys.privateKey));`
`var key = JSON.parse(sessionStorage.getItem("letsallprayitworks"));`

To get the key, certificate, generate the pkcs#12 and create a download link:
 
            var pem = <?php echo json_encode($this->certPem);?>;
			var pki = forge.pki;
			var cert = pki.certificateFromPem(pem);

			var p12Asn1 = forge.pkcs12.toPkcs12Asn1(key, cert, '1234', {algorithm: '3des'});
			var p12Der = forge.asn1.toDer(p12Asn1).getBytes();
			var p12b64 = forge.util.encode64(p12Der);

			var a = document.createElement('a');
			a.download = 'example.p12';
			a.setAttribute('href', 'data:application/x-pkcs12;base64,' + p12b64);
			a.appendChild(document.createTextNode('Download'));
			document.body.appendChild(a);

I'm attaching 2 files I've generated, one with AES, and one with 3DES

[pkcs12 files.zip](https://github.com/digitalbazaar/forge/files/443010/pkcs12.files.zip)

Thanks in advance!

**Comment by dlongley** [comment]
@pedrohls94,

Can you try taking the `sessionStorage` bit out of the picture and see if it helps? If you can't remove it for some reason, can you PEM-encode the private key before storing it?

**Comment by dlongley** [comment]
It may also be that the error is being raised because the database you're trying to *import into* has some problem. Have you tried this on a different machine?

**Comment by pedrohls94** [comment]
@dlongley,

So far I've tried PEM-encoding the private key before storing it, and still got the same error. Also, I have tried it on a different machine, no success.

I'm trying to find a solution which doesn't need the `sessionStorage` bit, to see if it solves the problem.

**Comment by dlongley** [comment]
@pedrohls94 -- If you're able to use the key after pulling it out of session storage then that's not the problem. I was poking around at the `SEC_ERROR_LEGACY_DATABASE` error and it seems like it's thrown in places unrelated to actually parsing the p12. It seems like it's related to the database you're trying to import it into... but without further exploration I can't be certain. Are you able to import other p12 containers, generated via other means, on the same machine into the same database?

**Comment by pedrohls94** [comment]
Apparently I did something wrong when testing it last time (maybe wrong password or whatever), I just know that now it works to import it in firefox/chrome.

I guess that what fixed my problem was the last change I made, this that you suggested:
> PEM-encode the private key before storing it

It's still giving out this same `SEC_ERROR_LEGACY_DATABASE` with pk12util, but since I only tried importing it there to see if it would give a better error message than firefox/chrome, which would only say "unknown error", it's not a problem to me.

Thank you kindly.
